src/Builder/ImportONNXUtils.cpp

```diff
@@ -0,0 +1,130 @@
+#include "ImportONNXUtils.hpp"
+#include <map>
+#include <set>
+#include <vector>
+
+bool TopologicallySorted(const onnx::GraphProto &graph) {
+  std::set<std::string> visited;
+  for (const auto &initializer : graph.initializer()) {
+    const auto &initializerName = initializer.name();
+    visited.insert(initializerName);
+  }
+  for (const auto &input : graph.input()) {
+    visited.insert(input.name());
+  }
+  for (const auto &node : graph.node()) {
+    for (const auto &input : node.input()) {
+      if (!visited.count(input))
+        return false;
+    }
+    for (const auto &output : node.output()) {
+      visited.insert(output);
+    }
+  }
+  return true;
+}
+
+// Lexicographically Smallest Topological Ordering
+bool SortGraph(onnx::GraphProto *graph) {
+  int nNodes = graph->node().size();
+  std::map<std::string, int> origIndex;
```

Comment:
Original Index is the index of the parent node of each edge. Since we are only considering edges that come from a node (because we grabbing output edges), we know that its "parent" idx is defined.

For the first iteration, the output edges come from the 0th indexed node.

Below is an illustration
```
   |   |                                     
 T1|   |T2    T1, T2 are inputs of Node Add  
   |   |      T3 is output of Node Add       
   +   +                                     
    Add                                      
     |                                       
  T3 |        T3 is input of Node Abs        
     |        T4 is output of Node Abs       
     +                                       
    Abs                                      
     |                                       
  T4 |                                       
     |                                                                          
  ```
The nodes are the ops, and the edges are the inputs and outputs. Generally we think of them as the tensors that result from ops.

I've also added comments to each of the vectors and sets, clarifying their purpose.